### PR TITLE
FSA-13 FSA-14 Prepare entities, dto and rest controller for cinema do…

### PR DESCRIPTION
**Pull request by KanekiSenapi**
…main

### CODE REVIEW

File: src/main/java/pl/aogiri/hhu/fsa/backend/cinema/domain/entity/CinemaFacilityEntity.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    @NotEmpty
+    private String name;
+
+    @NotEmpty
+    @Length(max = 1024)
+    private String description;
+
+    @NotEmpty
+    @Length(max = 128)
+    private String address;
```
Comment (ichal6) [inline src/main/java/pl/aogiri/hhu/fsa/backend/cinema/domain/entity/CinemaFacilityEntity.java:31]: Should this simple VARCHAR? Maybe shoudle we make separate table?

File: src/main/java/pl/aogiri/hhu/fsa/backend/cinema/web/controller/CinemaController.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            @ApiResponse(responseCode = "200", description = "Successful operation"),
+            @ApiResponse(responseCode = "404", description = "Cinema or facility not found")
+    })
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    @GetMapping(
+            value = "/{cinemaId}/facilities/{facilityId}",
+            produces = {MediaType.APPLICATION_JSON_VALUE}
+    )
+    default List<CinemaFacilityDto> getCinemaFacilityDetails(
```
Comment (ichal6) [inline src/main/java/pl/aogiri/hhu/fsa/backend/cinema/web/controller/CinemaController.java:80]: Should we return List of CinemaFacilityDTO? I think this method should return one object CinemaFacilityDTO.

**Comment by ichal6** [comment]
We have fields description, address and email in CinemaFacilityEntity, but we don't have in any DTOs. Should we create CinemaFacilityDetailsDto (with this fields)?

File: src/main/resources/db/migration/V2__create_tables_for_cinema.sql [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    name    varchar(64),
+    updated timestamp,
+    url     varchar(64),
+    primary key (id)
+);
+
+create table cinema_facility
+(
+    id          bigserial not null,
+    address     varchar(128),
```
Comment (ichal6) [inline src/main/resources/db/migration/V2__create_tables_for_cinema.sql:14]: I think we can separate address to another table. For example with latitude and longitude. And city field if we want to search cinema by city. Or is too complicated solution?

File: src/main/java/pl/aogiri/hhu/fsa/backend/cinema/domain/entity/CinemaEntity.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    @Id
+    private Long id;
+
+    @NotEmpty
+    @Length(max = 64)
+    private String name;
+
+    @Length(max = 64)
+    private String url;
```
Comment (ichal6) [inline src/main/java/pl/aogiri/hhu/fsa/backend/cinema/domain/entity/CinemaEntity.java:27]: Is it too short? Maybe some website has more than 64 chars?

Comment (KanekiSenapi) [reply src/main/java/pl/aogiri/hhu/fsa/backend/cinema/domain/entity/CinemaEntity.java:27]: As mvp, we want to provide data only for Poland, so the scope is cinemas such as helios, cinemacity, multikino. None of them exceed a length of 64, so it is not necessary to specify a longer length.

File: src/main/java/pl/aogiri/hhu/fsa/backend/cinema/domain/entity/CinemaFacilityEntity.java (hunk 1 above)
Comment (KanekiSenapi) [reply src/main/java/pl/aogiri/hhu/fsa/backend/cinema/domain/entity/CinemaFacilityEntity.java:31]: Specifying the length of the VARCHAR field allows for better database memory management, which can contribute to better database performance.

File: src/main/java/pl/aogiri/hhu/fsa/backend/cinema/web/controller/CinemaController.java (hunk 2 above)
Comment (KanekiSenapi) [reply src/main/java/pl/aogiri/hhu/fsa/backend/cinema/web/controller/CinemaController.java:80]: Yes

File: src/main/resources/db/migration/V2__create_tables_for_cinema.sql (hunk 3 above)
Comment (KanekiSenapi) [reply src/main/resources/db/migration/V2__create_tables_for_cinema.sql:14]: For mvp scope we don't need work on geographical or address data. If mvp scope will extend we will change it.

File: src/main/java/pl/aogiri/hhu/fsa/backend/cinema/domain/entity/CinemaEntity.java (hunk 4 above)
Comment (ichal6) [reply src/main/java/pl/aogiri/hhu/fsa/backend/cinema/domain/entity/CinemaEntity.java:27]: OK

File: src/main/java/pl/aogiri/hhu/fsa/backend/cinema/domain/entity/CinemaFacilityEntity.java (hunk 1 above)
Comment (ichal6) [reply src/main/java/pl/aogiri/hhu/fsa/backend/cinema/domain/entity/CinemaFacilityEntity.java:31]: OK

File: src/main/resources/db/migration/V2__create_tables_for_cinema.sql (hunk 3 above)
Comment (ichal6) [reply src/main/resources/db/migration/V2__create_tables_for_cinema.sql:14]: OK

**Review by ichal6 — APPROVED** [review]
I think it is ok now.

**Pull request merged**